src/Orchard.Tests/Tasks/DistributedLockServiceTests.cs

```diff
@@ -36,74 +36,64 @@ public class DistributedLockServiceTests : DatabaseEnabledTestsBase {
-            var attempt2 = _distributedLockService.TryAcquireLockForMachine(LockName, TimeSpan.FromSeconds(60), null, out @lock);
+            IDistributedLock @lock;
+            var attempt1 = _distributedLockService.TryAcquireLock(LockName, TimeSpan.FromSeconds(60), out @lock);
+            var attempt2 = _distributedLockService.TryAcquireLock(LockName, TimeSpan.FromSeconds(60), out @lock);
 
             Assert.That(attempt1, Is.True);
             Assert.That(attempt2, Is.True);
         }
 
-        [Test]
-        public void TryAcquiringLockTwiceOnSameMachineIncreasesLockCountTwice() {
-            DistributedLock @lock;
-            _distributedLockService.TryAcquireLockForMachine(LockName, TimeSpan.FromSeconds(60), null, out @lock);
-            _distributedLockService.TryAcquireLockForMachine(LockName, TimeSpan.FromSeconds(60), null, out @lock);
-
-            var lockId = Int32.Parse(@lock.Id);
-            var lockRecord = _distributedLockRepository.Get(lockId);
-            Assert.That(lockRecord.Count, Is.EqualTo(2));
-        }
 
         [Test]
-        public void ReleasingLockDecreasesLockCount() {
-            DistributedLock @lock;
-            _distributedLockService.TryAcquireLockForMachine(LockName, TimeSpan.FromSeconds(60), null, out @lock);
-            _distributedLockService.TryAcquireLockForMachine(LockName, TimeSpan.FromSeconds(60), null, out @lock);
+        public void AcquiringTheLockOnTheSameMachineReturnsTheSameLock() {
+            IDistributedLock lock1, lock2;
+            _distributedLockService.TryAcquireLock(LockName, TimeSpan.FromSeconds(60), out lock1);
+            _distributedLockService.TryAcquireLock(LockName, TimeSpan.FromSeconds(60), out lock2);
 
-            var lockId = Int32.Parse(@lock.Id);
-            var lockRecord = _distributedLockRepository.Get(lockId);
-            
-            _distributedLockService.ReleaseLock(@lock);
-            _session.Refresh(lockRecord);
-            Assert.That(lockRecord.Count, Is.EqualTo(1));
+            Assert.AreEqual(lock1, lock2);
         }
 
         [Test]
-        public void ReleasingLockAndCountReachesZeroDeletesLock()
-        {
-            DistributedLock @lock;
-            _distributedLockService.TryAcquireLockForMachine(LockName, TimeSpan.FromSeconds(60), null, out @lock);
+        public void ReleasingSingleLockDeletesRecord() {
+            IDistributedLock lock1;
+            _distributedLockService.TryAcquireLock(LockName, TimeSpan.FromSeconds(60), out lock1);
 
-            var lockId = Int32.Parse(@lock.Id);
-            _distributedLockService.ReleaseLock(@lock);
-            var lockRecord = _distributedLockRepository.Get(lockId);
+            lock1.Dispose();
+            var lockRecord = _distributedLockRepository.Table.FirstOrDefault();
 
-            Assert.That(lockRecord, Is.Null);
+            Assert.IsNull(lockRecord);
```

Comment:
I prefer Assert.That.... But that's just me.

-----Original Message-----
From: "Sébastien Ros" <notifications@github.com>
Sent: ‎03/‎09/‎2015 17:02
To: "OrchardCMS/Orchard" <Orchard@noreply.github.com>
Subject: Re: [Orchard] Refactoring locking strategy (#5734)

In src/Orchard.Tests/Tasks/DistributedLockServiceTests.cs:
>  
> -            Assert.That(lockRecord, Is.Null);
> +            Assert.IsNull(lockRecord);
Was not even intentional ;) Same.
—
Reply to this email directly or view it on GitHub.